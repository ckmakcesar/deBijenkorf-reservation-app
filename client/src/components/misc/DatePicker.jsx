import React, { forwardRef } from 'react';
import StandardDatePicker from "react-datepicker";
import PropTypes from 'prop-types';

import "react-datepicker/dist/react-datepicker.css";
import '../../styles/misc/DatePicker.css'; // override default stylesheet

const DatePicker = forwardRef(({
  label,
  required,
  dateFormat,
  selected,
  onChange,
  filterDate,
  divId,
}, ref) => (
  <label className='input-field-label'>
    <div>{label ? `${label}${label && required ? '*' : ''}` : undefined}</div>

    <div
      {...(divId ? { id: `DatePickerDiv-${divId}` } : null)} // optional id - should be unique
      className='customDatePickerWidth'
    >
      <StandardDatePicker
        ref={ref}
        dateFormat={dateFormat}
        selected={selected}
        onChange={onChange}
        filterDate={filterDate}
      />
    </div>
  </label>
));

// *********************************************************************************************************************
// PROPS
// *********************************************************************************************************************
DatePicker.propTypes = {
  label: PropTypes.string.isRequired,
  required: PropTypes.bool,
  dateFormat: PropTypes.string,
  selected: PropTypes.object,
  onChange: PropTypes.func.isRequired,
  filterDate: PropTypes.func,
  divId: PropTypes.string,
};

DatePicker.defaultProps = {
  required: false,
  dateFormat: 'dd/MM/yyyy',
  selected: null,
  filterDate: () => { },
  divId: '',
};

DatePicker.displayName = 'DatePicker';

export default DatePicker;
