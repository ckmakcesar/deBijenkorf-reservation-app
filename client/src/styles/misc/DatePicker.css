.customDatePickerWidth,
.customDatePickerWidth > div.react-datepicker-wrapper,
.customDatePickerWidth > div > div.react-datepicker__input-container {
  width: 100%;
}
.customDatePickerWidth > div > div.react-datepicker__input-container input {
  width: calc(100% - 8px);
}
