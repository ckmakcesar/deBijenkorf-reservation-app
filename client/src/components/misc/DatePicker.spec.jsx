import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import DatePicker from './DatePicker';

describe('component:DatePicker', () => {
  it('Should render DatePicker with every optional props absent', () => {
    const wrapper = shallow(
      <DatePicker
        label='Choose a date below'
        onChange={jest.fn()}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render DatePicker with every optional props present', () => {
    const wrapper = shallow(
      <DatePicker
        label='Choose a date below'
        required={true}
        dateFormat='yyyy/MM/dd'
        selected={null}
        onChange={jest.fn()}
        filterDate={(t) => (new Date(t) > new Date())}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
