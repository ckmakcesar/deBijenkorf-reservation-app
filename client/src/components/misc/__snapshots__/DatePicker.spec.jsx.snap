// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:DatePicker Should render DatePicker with every optional props present 1`] = `
<label
  className="input-field-label"
>
  <div>
    Choose a date below*
  </div>
  <div
    className="customDatePickerWidth"
  >
    <r
      allowSameDay={false}
      customTimeInput={null}
      dateFormat="yyyy/MM/dd"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      enableTabLoop={true}
      excludeScrollbar={true}
      filterDate={[Function]}
      focusSelectedMonth={false}
      monthsShown={1}
      nextMonthAriaLabel="Next Month"
      nextMonthButtonLabel="Next Month"
      nextYearAriaLabel="Next Year"
      nextYearButtonLabel="Next Year"
      onBlur={[Function]}
      onCalendarClose={[Function]}
      onCalendarOpen={[Function]}
      onChange={[MockFunction]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      preventOpenOnFocus={false}
      previousMonthAriaLabel="Previous Month"
      previousMonthButtonLabel="Previous Month"
      previousYearAriaLabel="Previous Year"
      previousYearButtonLabel="Previous Year"
      readOnly={false}
      renderDayContents={[Function]}
      selected={null}
      selectsDisabledDaysInRange={false}
      shouldCloseOnSelect={true}
      showFourColumnMonthYearPicker={false}
      showFullMonthYearPicker={false}
      showMonthYearPicker={false}
      showPopperArrow={true}
      showPreviousMonths={false}
      showQuarterYearPicker={false}
      showTimeInput={false}
      showTimeSelect={false}
      showTwoColumnMonthYearPicker={false}
      showYearPicker={false}
      strictParsing={false}
      timeCaption="Time"
      timeInputLabel="Time"
      timeIntervals={30}
      withPortal={false}
      yearItemNumber={12}
    />
  </div>
</label>
`;

exports[`component:DatePicker Should render DatePicker with every optional props absent 1`] = `
<label
  className="input-field-label"
>
  <div>
    Choose a date below
  </div>
  <div
    className="customDatePickerWidth"
  >
    <r
      allowSameDay={false}
      customTimeInput={null}
      dateFormat="dd/MM/yyyy"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      enableTabLoop={true}
      excludeScrollbar={true}
      filterDate={[Function]}
      focusSelectedMonth={false}
      monthsShown={1}
      nextMonthAriaLabel="Next Month"
      nextMonthButtonLabel="Next Month"
      nextYearAriaLabel="Next Year"
      nextYearButtonLabel="Next Year"
      onBlur={[Function]}
      onCalendarClose={[Function]}
      onCalendarOpen={[Function]}
      onChange={[MockFunction]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      preventOpenOnFocus={false}
      previousMonthAriaLabel="Previous Month"
      previousMonthButtonLabel="Previous Month"
      previousYearAriaLabel="Previous Year"
      previousYearButtonLabel="Previous Year"
      readOnly={false}
      renderDayContents={[Function]}
      selected={null}
      selectsDisabledDaysInRange={false}
      shouldCloseOnSelect={true}
      showFourColumnMonthYearPicker={false}
      showFullMonthYearPicker={false}
      showMonthYearPicker={false}
      showPopperArrow={true}
      showPreviousMonths={false}
      showQuarterYearPicker={false}
      showTimeInput={false}
      showTimeSelect={false}
      showTwoColumnMonthYearPicker={false}
      showYearPicker={false}
      strictParsing={false}
      timeCaption="Time"
      timeInputLabel="Time"
      timeIntervals={30}
      withPortal={false}
      yearItemNumber={12}
    />
  </div>
</label>
`;
